### PR TITLE
1579 refactoring of last comments

### PR DESCRIPTION
**Pull request by psiotwo**
https://kbss.felk.cvut.cz/redmine/issues/1579

### CODE REVIEW

File: src/component/dashboard/widget/lastcommented/CommentedAssetList.tsx [hunk 1]
```diff
@@ -58,36 +59,42 @@ export const CommentedAssetList: React.FC<CommentedAssetListProps> = (
     [formatMessage, user, locale]
   );
 
+    const renderComment = useCallback(
```
Comment (ledsoft) [inline src/component/dashboard/widget/lastcommented/CommentedAssetList.tsx]: Doesn't seem like being formatted according to the Prettier rules.

File: src/model/RecentlyCommentedAsset.ts [hunk 2]
```diff
@@ -15,17 +16,20 @@ export interface RecentlyCommentedAssetData extends HasTypes {
     iri: string;
     type: string;
     lastComment: Comment;
+    lastMyComment?: Comment;
```
Comment (ledsoft) [inline src/model/RecentlyCommentedAsset.ts]: Same as on the  backend: why not `myLastComment`?

File: src/model/RecentlyCommentedAsset.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ const ctx = {
     editor: VocabularyUtils.PREFIX + "má-editora",
     vocabulary: VocabularyUtils.IS_TERM_FROM_VOCABULARY,
     lastComment: VocabularyUtils.JE_TEMATEM,
+    lastMyComment: VocabularyUtils.MA_MUJ_POSLEDNI_KOMENTAR,
```
Comment (ledsoft) [inline src/model/RecentlyCommentedAsset.ts]: Context property does not match the property defined in the `RecentlyCommentedAssetData` interface.

**Review by ledsoft — APPROVED** [review]
LGTM. Can't check the functionality, I assume backend must be available as well.

**Comment by psiotwo** [comment]
> LGTM. Can't check the functionality, I assume backend must be available as well.

Yes, I will merge backend to dev first before closing this one and ping you again.

**Comment by psiotwo** [comment]
> > LGTM. Can't check the functionality, I assume backend must be available as well.
> 
> Yes, I will merge backend to dev first before closing this one and ping you again.

@ledsoft backend is deployed, You can check deploy-preview. In order to test it you would need to have two users - one who creates a comment and the other who reacts to it.

**Review by MichalMed — COMMENTED** [review]
1. I see comments of other users in "Mine". Is this expected behaviors?
2. "With reaction to my comment" contains my last comment plus last comment at all. If the last comment is also mine, it does not show anything at all. Is this expected behavior?
3. When I click on term label in the Last commented terms block, I would expect to be redirected to comments tab of that term.

**Comment by psiotwo** [comment]
> 1. I see comments of other users in "Mine". Is this expected behaviors?

Yes, that are my terms (not my comments), i.e. terms I created/edited.

> 2. "With reaction to my comment" contains my last comment plus last comment at all. If the last comment is also mine, it does not show anything at all. Is this expected behavior?

Was thinking about this as well, and I think it is correct - if both comments are the same then the first one cannot be a reaction to the other one. I would pass to IPR for testing to clarify the semantics if needed.

> 3. When I click on term label in the Last commented terms block, I would expect to be redirected to comments tab of that term.

Thanks for reminder. Fixed

**Review by MichalMed — APPROVED** [review]
LGBT

**Pull request merged**